cranelift-wasm/src/module_translator.rs

```diff
@@ -137,6 +137,16 @@ pub fn translate_module<'data>(
     if let SectionCode::Data = section.code {
         let data = section.get_data_section_reader()?;
         parse_data_section(data, environ)?;
+
+        if reader.eof() {
+            return Ok(());
+        }
+        section = reader.read()?;
+    }
+
+    if let SectionCode::Custom { name: "name", .. } = section.code {
```

Comment:
Can you match on `kind: CustomSectionKind::Name` instead of `name: "name"`?

Reply:
Fixed this, good eye!